src/sim_recon/images/utils.py

```diff
@@ -12,33 +14,94 @@
 logger = logging.getLogger(__name__)
 
 
+def _calculate_shape_crop_slices(
+    target_xy_shape: NDArray[np.integer[Any]],
+    current_xy_shape: NDArray[np.integer[Any]],
+    centre: NDArray[np.integer[Any]],
+) -> tuple[slice, slice]:
+    if np.any(target_xy_shape > current_xy_shape):
+        raise OutOfBoundsError("Target crop shape exceeds at least one array dimension")
+
+    from_centre = (target_xy_shape - 1) / 2
+    min_bounds = np.ceil(centre - from_centre).astype(np.uint16)
+    max_bounds = np.ceil(
+        centre + from_centre + 1  # slices are not inclusive of top bound
+    ).astype(np.uint16)
+    if np.any(min_bounds < 0) or np.any(max_bounds > current_xy_shape):
+        raise OutOfBoundsError("Target crop extends beyond the bounds of the image")
+    return slice(min_bounds[0], max_bounds[0]), slice(
+        min_bounds[1], max_bounds[1]
+    )  # x, y
+
+
+def _calculate_decimal_crop_slices(
+    crop: float,
+    current_xy_shape: NDArray[np.integer[Any]],
+    centre: NDArray[np.integer[Any]],
+) -> tuple[slice, slice]:
+    if crop <= 0 or crop >= 1:
+        raise InvalidValueError("Argument 'crop' must be >= 0 and < 1")
+    from_centre = (current_xy_shape * (1 - crop) - 1) / 2
+    min_bounds = np.ceil(centre - from_centre).astype(np.uint16)
+    max_bounds = np.ceil(
+        centre + from_centre + 1  # slices are not inclusive of top bound
+    ).astype(np.uint16)
+    if np.any(min_bounds < 0) or np.any(max_bounds > current_xy_shape):
+        raise OutOfBoundsError("Target crop extends beyond the bounds of the image")
+    return slice(min_bounds[0], max_bounds[0]), slice(
+        min_bounds[1], max_bounds[1]
+    )  # x, y
+
+
 def apply_crop(
     array: NDArray[Any],
     xy_shape: tuple[int, int] | None = None,
     crop: float = 0,
+    xy_centre: tuple[int, int] | None = None,
 ) -> NDArray[Any]:
+
     assert array.ndim > 1, "Images must have > 1 dimension"
+
     array_slices = [slice(None)] * array.ndim
+
+    current_xy_shape = np.asarray((array.shape[-1], array.shape[-2]), dtype=np.uint16)
```

Comment:
Seems like your code is contorted by preserving the x-y ordering rather transposing to the storage ordering